docs/conditional-gatherer/development.md

```diff
@@ -0,0 +1,181 @@
+# Conditional Gatherer Development
+
+## Validation of the conditional rules
+
+The Insights Operator internally validates the conditional rules JSON against the JSON schema. [The schema](../../pkg/gatherers/conditional/gathering_rules.schema.json) is available in the `pkg/gatherers/conditional` package. You can see that this schema refers to the second [gathering_rule.schema.json](../../pkg/gatherers/conditional/gathering_rule.schema.json). This second schema defines the more important restrictions on the specific rules. **If the validation fails, no conditional data will be gathered!**!
```

Comment:
> If the validation fails, no conditional data will be gathered!

Does this mean that an error in one gathering rule prevents all other gathering rules from being executed?

Reply:
Yep. We can address it on a separate task, but sure, we can change it. I'll probably drag this as a 'next step' point of improvement, since we may also need to track the failings conditions somehow.